hw/ip/flash_ctrl/doc/_index.md

```diff
@@ -65,40 +75,130 @@ The physical controller supports the following features
-Importantly, the flash protocol controller is not responsible for the detailed timing and waveform control of the flash.
-Instead, it maintains FIFOs / interrupts for the software to process data, as well as high level abstraction of region protection controls.
+The Flash Protocol Controller sits between the host software interface, other hardware components and the flash physical controller.
+Its primary functions are two fold
+*  Translate software program, erase and read requests into a high level protocol for the actual flash physical controller
+*  Act as communication interface between flash and other components in the system, such as life cycle and key manager.
+
+Importantly, the flash protocol controller is not responsible for the detailed timing and waveform control of the flash, nor is it responsible for data scrambling and reliability metadata such as parity and ECC.
+Instead, it maintains FIFOs / interrupts for the software to process data, as well as high level abstraction of region protection controls as well as error handling.
+
+There are two hardware connections between the flash controller and the rest of the design.
+*  The first is to life cycle, in which flash content plays a part.
+*  The second is to key manager, whose seed material may be partially sourced from flash.
 
 ### Flash Physical Controller
 
-The Flash Physical Controller is the wrapper module that contains the actual vendor flash memory instantiation.
-It is responsible for arbitrating high level protocol commands (such as read, program, erase) as well as any additional security and reliability features.
+The Flash Physical Controller is the wrapper module that contains the actual flash memory instantiation.
+It is responsible for arbitrating high level protocol commands (such as read, program, erase) as well as any additional security (scrambling) and reliability (ECC) features.
 The contained vendor wrapper module is then responsible for converting high level commands into low level signaling and timing specific to a particular flash vendor.
 The vendor wrapper module is also responsible for any BIST, redundancy handling, remapping features or custom configurations required for the flash.
 
-The diagram below summarizes the high level breakdown.
+The flash physcial controller supports the flash scramble function.
+The scramble keys are provided from an external static block such as the OTP.
+
+### Flash Scrambling
+
+Flash scrambling is built using the [XEX tweakable block cipher](https://en.wikipedia.org/wiki/Disk_encryption_theory#Xor%E2%80%93encrypt%E2%80%93xor_(XEX)).
+
+When a read transaction is sent to flash, the following steps are taken:
+*  The tweak is calculated using the transaction address and a secret address key through a galois multiplier.
+*  The data content is read out of flash.
+*  If the data content is scrambled, the tweak is XOR'd with the scrambled text and then decrypted through the prince block cipher using a secret data key.
```

Comment:
If I programmed the flash with scrambling disabled, then enabled scrambling and read back the same data, how does the logic know to skip the descrambling?

Reply:
it would not. The user has to make this consistent.  So if you scramble an area, then mark it as not scramble disable and attempt to read it back, the data will be garbage.  The same is true of the reverse direction.

Reply:
Got it, thanks.

Reply:
added more text to explain.